app/src/main/java/org/oppia/android/app/devoptions/marktopicscompleted/MarkTopicsCompletedFragmentPresenter.kt

```diff
@@ -64,15 +64,22 @@ class MarkTopicsCompletedFragmentPresenter @Inject constructor(
     }
 
     binding.markTopicsCompletedAllCheckBoxContainer.setOnClickListener {
-      if (binding.isAllChecked == null || binding.isAllChecked == false)
+      if (binding.isAllChecked == null || binding.isAllChecked == false) {
         binding.isAllChecked = true
+      } else if (binding.isAllChecked == true) {
+        binding.isAllChecked = false
+      }
```

Comment:
Given that this covers all cases, this can probably be simplified to be more Kotlin idiomatic:

```kotlin
binding.isAllChecked = !(binding.isAllChecked ?: false)
```

Ditto above in the PR.

Reply:
This appears resolved in the latest changes.